flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/DefaultSlotPoolServiceSchedulerFactory.java

```diff
@@ -223,15 +223,15 @@ public static DefaultSlotPoolServiceSchedulerFactory fromConfiguration(
     }
 
     private static JobManagerOptions.SchedulerType getSchedulerType(
-            Configuration configuration, JobType jobType, boolean isDynamic) {
-        if (configuration.get(JobManagerOptions.SCHEDULER_MODE)
-                == SchedulerExecutionMode.REACTIVE) {
+            Configuration configuration, JobType jobType, boolean isDynamicGraph) {
+        if (configuration.get(JobManagerOptions.SCHEDULER_MODE) == SchedulerExecutionMode.REACTIVE
+                || System.getProperties().containsKey("flink.tests.enable-adaptive-scheduler")) {
```

Comment:
This is not consistent with `ClusterOptions#isAdaptiveSchedulerEnabled()`, which checks system properties only if the SCHEDULER_MODE is not set.